packages/web-wallet/src/services/TokenStorageService.ts

```diff
@@ -0,0 +1,164 @@
+import type { TokenInfo, TokenStorageData } from "../types/token";
+
+/**
+ * Service for persisting token registrations to localStorage with network-specific keys.
+ *
+ * Storage format:
+ * Key: `hathor_wallet_registered_tokens_{network}_{genesisHash}`
+ * Value: JSON serialized TokenStorageData
+ *
+ * Note: genesisHash is currently empty string, will be populated when RPC handler is updated
+ */
+export class TokenStorageService {
```

Comment:
Maybe this service should have an in-memory copy and keep the in-memory and localStorage synced.
in-memory fetches are way faster.

Reply:
It could, but we already have a in-memory copy in the state so read operations are already very fast:

```
  localStorage (TokenStorageService)
         ↓ load once at startup
  React state (registeredTokens)
         ↓ used for all reads
  UI components / refreshBalance / etc
```